### ISSUE TITLE
Inconsistency Found in Reconstruction loss

### ISSUE THREAD

**Issue by atul04** [iesl/diora#8]
Hi, in paper its been suggested that Authors had done their experiments on the reconstruction loss on all nodes of the tree but in their code base they did an agreement at word level only with the elmo embeddings. Can you please look into it once.

**Comment by TRUMANCFY** [comment]
Also, for the word level reconstruction, the outputs should not be all-zeros, right? The first position should be one, corresponding to the positive inputs?

https://github.com/iesl/diora/blob/4691b7c1cc900e2b9b9872d5e7194338eb13341d/pytorch/diora/net/trainer.py#L65

**Comment by mrdrozdov** [comment]
The code is consistent with the paper.

We train DIORA to predict a word conditioned on the sentence context (minus that word) and a soft encoding over all valid constituency trees.

We say as much in the paper (Section 2.3 Training Objective):

> The terminal outside vector $\bar{b}(i)$ is trained to predict its original input $v_i$.

Paper: https://aclanthology.org/N19-1116.pdf
